### ISSUE TITLE
Do a version 1.0.0 release

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/NCEPLIBS-ncio#10]
We need to do an initial release. I would like to see documentation and testing working first.

**Comment by jswhit** [comment]
I have a [test program](https://github.com/jswhit/fv3gfs_ncio/blob/master/test_fv3gfs_ncio.f90) that could be adapted for github actions.

**Comment by edwardhartnett** [comment]
Yes please!

**Comment by jswhit** [comment]
Is there a fortran unit testing framework as part of NCEPLIBS?  Does such a thing even exist?

**Comment by edwardhartnett** [comment]
No, there is not, and will not be. There are some, but they are worthless.

Just do what is done in netcdf-fortran: write a small fortran program to create a file, put some stuff in it, close it. Then reopen and check. For any failure, stop 2.

No command line arguments or anything fancy, just a simple f90 program that creates a test file in one way, and then checks it for correctness.

If you want to wait I'll take a stab at something after I've finished some super-annoying problems I'm currently having with the ESMF build in github actions...

**Comment by kgerheiser** [comment]
I can take your test program and setup the testing CMake real fast.

**Comment by edwardhartnett** [comment]
OK, we've got basic testing and documentation. Let's do a 1.0.0 release.

@Hang-Lei-NOAA can you do this? It should include:
1 - Update VERSION file.
2 - Write release notes on github.
3 - Build documentation and publish to gh-pages.
4 - Mark and release in GitHub.
5 - Announce release.

**Comment by edwardhartnett** [comment]
@Hang-Lei-NOAA can we do a 1.0.0 release this week?

**Comment by Hang-Lei-NOAA** [comment]
Sure, that works for me.

**Comment by edwardhartnett** [comment]
I mean, can you do it please. ;-)

**Comment by Hang-Lei-NOAA** [comment]
Yes, I can.

**Comment by edwardhartnett** [comment]
@Hang-Lei-NOAA  I have added release notes. Please build the documentation and upload it to the gh-pages branch. Let me know if you need help.

**Comment by Hang-Lei-NOAA** [comment]
Hi, Ed, which gh-pages that we need to post? I was trying to push to
NOAA-EMC/NOAA-EMC.github.io. Not sure if it is right. That site has many
problems. I am also trying to fix.

**Comment by edwardhartnett** [comment]
OK good!

Are you the one in charge of NOAA-EMC/NOAA-EMC.github.io?

As for NCEPLIBS-ncio, here's how you do it:
* Check out develop branch, build the documentation.
* Elsewhere, clone the repo again, into a different directory.
* In that directory: 

```
git branch gh-pages
git checkout gh-pages
git rm -rf *
cp -R ~/NCEPLIBS-ncio/build/docs/html/* .
git add *
git commit -m "updated docs" -a
git push

```

Basically you create a branch called gh-pages, you delete everything on it, you copy in the built HTML documentation, and add and commit that to the gh-pages branch.

When complete, the gh-pages branch does not contain any code, it contains the html output of the doxygen build. That will be automatically displayed on the .io site by GitHub. (It may take a few minutes to appear after you do the push - it takes time for GitHub to update servers.)

Let me know if you have questions.

**Comment by Hang-Lei-NOAA** [comment]
hi, Ed, I am not the manager of that repo, but I have asked them to add you
as the manager of that repo. No branch is allowed to be added to that repo.
I have made a copy to my personal gh-pages.
Also I made a fork, but the fork cannot be shown following github pages
policy. So, I will use my personal account gh-page to show, but use the
fork to push back.
If the pull is not passed on time, I will use the official address in the
release. But use my personal account gh-pages as the validation.
